### PR TITLE
Fix/bugs

### LINKED ISSUE #95: Android AccountsTab ScrollView does not scroll
untitled

### LINKED ISSUE #100: Android screens flash white on transitions
Even after #93, screen still flashes white when moving between some screens

### LINKED ISSUE #122: Change Delete Wallet UI
For the delete wallet pop-up, “Delete” should be on the right side, and “Cancel” on the left side. “Delete” should also be marked in red, as it’s a destructive action. Title: “Do you want to delete the wallet <WALLET_NAME>?”, description could be “You cannot do undo this action”.

Maybe we can use the new blurry modal UI instead of the native dialog.

### LINKED ISSUE #127: Android weird input effect on seed words
Trying to type `text` causes this:

https://user-images.githubusercontent.com/24814046/166157464-e9a73b12-d662-49cd-be26-4cd093569509.mp4

### LINKED ISSUE #128: Android Loading UI is broken
https://user-images.githubusercontent.com/24814046/166256640-ce90ca80-5d52-4ef4-a310-0c489427bf59.mp4

### LINKED ISSUE #130: Android Loading Splash Icon Blur
image

### LINKED ISSUE #133: Android Loading Modal Align Center Bug

### PR DESCRIPTION
**Pull request by tsejerome**
Resolves #128, Resolves #127, Resolves #122, Resolves #100, Resolves #133, Resolves #130, Resolves #95

### CODE REVIEW

**Comment by aguycalled** [comment]
About the “Delete Wallet” view:
- Good to add more margin on the sides
- Buttons do not indicate (as in changing their color) when touched
- Users would expect to be able to dismiss a blurred view by tapping anywhere on the view itself (apart from the buttons)

**Comment by aguycalled** [comment]
About the “Delete Wallet” functionality:
- After clicking “Delete”, it dismissed the view. I was left on the “Settings” view for 1-2 seconds before it took me back to the “Whisper” screen. I had time to select something else from the Settings list in the meanwhile. Maybe best to show a “Deleting wallet” screen when tapping “Delete” similar to the “Loading wallet” screen?
- After deleting my wallet, I was taken back to the “Whisper” screen, and I could still tap “Open wallet”, in which I found the same wallet I just deleted. I tapped it, and then it took around 45 seconds to open it. After it opened, I was to delete the wallet again. I tried doing this 4-5 times, and on the 3rd or 4th try, it gave me an error (and then proceeded to take me back to the “Whisper” screen)

Can be fixed with this patch:

```patch
diff --git a/contexts/WalletContext.tsx b/contexts/WalletContext.tsx
index 7af4f25..4152a06 100755
--- a/contexts/WalletContext.tsx
+++ b/contexts/WalletContext.tsx
@@ -28,6 +28,7 @@ export interface WalletContextValue {
   updateAccounts: () => void;
   bootstrapProgress: number;
   firstSyncCompleted: boolean;
+  removeWallet: () => void;
 }
 
 export const WalletContext = createContext<WalletContextValue | undefined>(
diff --git a/contexts/WalletProvider.tsx b/contexts/WalletProvider.tsx
index a4491ea..7b07843 100755
--- a/contexts/WalletProvider.tsx
+++ b/contexts/WalletProvider.tsx
@@ -479,6 +479,13 @@ export const WalletProvider = (props: any) => {
     }
   }, [wallet]);
 
+  const removeWallet = useCallback(async (name: string) => {
+    if (njs) {
+      await njs.wallet.WalletFile.RemoveWallet(name);
+      njs.wallet.WalletFile.ListWallets().then(setWalletsList);
+    }
+  }, [njs]);
+
   const walletContext: WalletContextValue = useMemo(
     () => ({
       bitcore: njs?.wallet.bitcore,
@@ -506,6 +513,7 @@ export const WalletProvider = (props: any) => {
       nfts,
       updateAccounts,
       bootstrapProgress,
+      removeWallet
     }),
     [
       win,
@@ -527,6 +535,7 @@ export const WalletProvider = (props: any) => {
       nfts,
       updateAccounts,
       bootstrapProgress,
+      removeWallet
     ],
   );
 
diff --git a/screens/wallet/SettingsScreen.tsx b/screens/wallet/SettingsScreen.tsx
index 17949ee..7b5271d 100644
--- a/screens/wallet/SettingsScreen.tsx
+++ b/screens/wallet/SettingsScreen.tsx
@@ -30,7 +30,7 @@ interface SettingsItem {
 const SettingsScreen = (props: ScreenProps<'SettingsScreen'>) => {
   const {readPassword} = useSecurity();
   const [loading, setLoading] = useState<string | undefined>(undefined);
-  const {walletName, wallet, createWallet} = useWallet();
+  const {walletName, wallet, createWallet, removeWallet} = useWallet();
   const {njs} = useNjs();
   const bottomSheet = useBottomSheet();
   const {changeMode, supportedType, currentAuthenticationType} = useSecurity();
@@ -115,8 +115,9 @@ const SettingsScreen = (props: ScreenProps<'SettingsScreen'>) => {
     wallet.Disconnect();
     wallet.CloseDb();
     if (deleteWallet) {
-      await njs.wallet.WalletFile.RemoveWallet(walletName);
+      await removeWallet(walletName);
     }
+    setLoading(undefined);
     navigate('Intro');
   };
 
@@ -129,10 +130,14 @@ const deleteWallet = () => {
       [
         {
           text: 'Delete',
-          onPress: () => {
-            readPassword().then((password: string) => {
-              disconnectWallet(true);
-            });
+          onPress: async () => {
+            setLoading('Deleting...');
+            try {
+              await readPassword();
+              await disconnectWallet(true);
+            } catch (e) {
+              setLoading(undefined);
+            }
           },
         },
         {

```

**Comment by aguycalled** [comment]
both tapping the menu hamburger or the refresh do nothing on android and cause this:

```
ReferenceError: Property 'Connection_Stats_Enum' doesn't exist
ReferenceError: Property 'Connection_Stats_Enum' doesn't exist
    at _callee$ (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:272945:32)
    at call (native)
    at tryCatch (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26336:23)
    at invoke (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26506:32)
    at anonymous (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26378:30)
    at call (native)
    at tryCatch (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26336:23)
    at invoke (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26408:30)
    at anonymous (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26438:19)
    at tryCallTwo (/tmp/hermes/staging/hermes/cmake/intlDebug/arm64-v8a/lib/InternalBytecode/InternalBytecode.js:61:9)
    at doResolve (/tmp/hermes/staging/hermes/cmake/intlDebug/arm64-v8a/lib/InternalBytecode/InternalBytecode.js:216:25)
    at Promise (/tmp/hermes/staging/hermes/cmake/intlDebug/arm64-v8a/lib/InternalBytecode/InternalBytecode.js:82:14)
    at callInvokeWithMethodAndArg (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26437:33)
    at enqueue (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26442:157)
    at anonymous (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26378:30)
    at anonymous (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:26457:69)
    at _callee (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:272941:40)
    at anonymous (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:76268:55)
    at apply (native)
    at invokeGuardedCallbackProd (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:4769:21)
    at apply (native)
    at invokeGuardedCallback (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:4873:42)
    at apply (native)
    at invokeGuardedCallbackAndCatchFirstError (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:4877:36)
    at executeDispatch (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:4955:48)
    at executeDispatchesInOrder (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:4975:26)
    at executeDispatchesAndRelease (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:6235:35)
    at executeDispatchesAndReleaseTopLevel (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:6244:43)
    at call (native)
    at forEachAccumulated (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:5508:18)
    at runEventsInBatch (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:6259:27)
    at runExtractedPluginEventsInBatch (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:6339:25)
    at anonymous (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:6314:42)
    at batchedUpdates$1 (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:18376:20)
    at batchedUpdates (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:6221:36)
    at _receiveRootNodeIDEvent (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:6313:23)
    at receiveEvent (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:6343:32)
    at apply (native)
    at __callFunction (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:24663:36)
    at anonymous (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:24391:31)
    at __guard (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:24617:15)
    at callFunctionReturnFlushedQueue (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:24390:21)
 WARN  Please report: Excessive number of pending callbacks: 501. Some pending callbacks that might have leaked by never being called from native code: {"665":{},"701":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"727":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"945":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"967":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"989":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"1011":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"1040":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"4874":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5009":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5019":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5053":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5092":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5128":{},"5160":{},"5192":{},"5224":{},"5256":{},"5288":{},"5320":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5353":{},"5387":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5420":{},"5452":{},"5484":{},"5516":{},"5548":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5584":{},"5616":{},"5648":{},"5680":{},"5712":{},"5744":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"5780":{},"5812":{},"5844":{},"5876":{},"5908":{},"5940":{},"5972":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"6008":{},"6040":{},"6072":{},"6104":{},"6136":{},"6168":{"module":"ReactNativeExceptionHandler","method":"setHandlerforNativeException"},"6204":{},"6236":{},"6268":{},"6300":{},"...(truncated keys)...":451} 
    at App (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:114608:22)
    at ModalProvider (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:403582:38)
    at RCTView
    at View
    at ModalPanel (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:183657:22)
    at ThemeProvider (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:183488:22)
    at MappingProvider (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:183404:22)
    at StyleProvider (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:183342:22)
    at ApplicationProvider (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:181487:22)
    at RNCSafeAreaProvider
    at SafeAreaProvider (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:115545:24)
    at AppWrapper (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:114857:50)
    at RCTView
    at View
    at RCTView
    at View
    at AppContainer (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:81148:36)
    at Whisper(RootComponent) (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false&app=com.navcash&modulesOnly=false&runModule=true:88047:28)
```

**Comment by aguycalled** [comment]
This patch would solve the previous issue:

```patch
diff --git a/App.tsx b/App.tsx
index 4323546..d930fd5 100755
--- a/App.tsx
+++ b/App.tsx
@@ -87,25 +87,24 @@ const App = (props: {theme: string}) => {
       openModal(<ErrorModalContent errorText={errorMsg}></ErrorModalContent>);
     });
   };
-
-  const JSLeveErrorPrompt = async (error: Error | string, isFatal: boolean) => {
-    await saveGlobalErrorRecord(errorTextParser(error, isFatal));
-    await saveTemporaryErrorRecord(errorTextParser(error, isFatal));
-    closeModal();
-    promptErrorToaster(error, isFatal, false, () => {
-      const errorMsg = errorTextParser(error, isFatal);
-      openModal(<ErrorModalContent errorText={errorMsg}></ErrorModalContent>);
+  
+  useEffect(() => {
+    setJSExceptionHandler(async (error: Error | string, isFatal: boolean) => {
+      await saveGlobalErrorRecord(errorTextParser(error, isFatal));
+      await saveTemporaryErrorRecord(errorTextParser(error, isFatal));
+      closeModal();
+      promptErrorToaster(error, isFatal, false, () => {
+        const errorMsg = errorTextParser(error, isFatal);
+        openModal(<ErrorModalContent errorText={errorMsg}></ErrorModalContent>);
+      });
+    }, true);
+
+    setNativeExceptionHandler(async errorString => {
+      await saveGlobalErrorRecord(errorTextParser(errorString, true));
+      await saveTemporaryErrorRecord(errorTextParser(errorString, true));
+      await AsyncStorage.setItem('crashErrorRecords', errorString);
     });
-  };
-  setJSExceptionHandler(JSLeveErrorPrompt, true);
-
-  setNativeExceptionHandler(async errorString => {
-    await saveGlobalErrorRecord(errorTextParser(errorString, true));
-    await saveTemporaryErrorRecord(errorTextParser(errorString, true));
-    await AsyncStorage.setItem('crashErrorRecords', errorString);
-  });
 
-  useEffect(() => {
     AsyncStorage.getItem('shownWelcome').then(itemValue => {
       if (itemValue == 'true' || itemValue == 'false') {
         setShownWelcome(itemValue);
```

**Comment by tsejerome** [comment]
Oh I haven't pushed the fix on my branch regarding the connection state, it's an import issue

**Comment by aguycalled** [comment]
ok, but I think there is also a loop with `setJSExceptionHandler` and `setNativeExceptionHandler` which is solved if they are wrapped in `useEffect`

File: screens/wallet/SettingsScreen.tsx [hunk 1]
```diff
@@ -184,6 +175,12 @@ const SettingsScreen = (props: ScreenProps<'SettingsScreen'>) => {
       show: supportedType != SecurityAuthenticationTypes.MANUAL,
       onPress: () => biometricsAlert(),
     },
+    {
+      title: 'Biometrics check',
```
Comment (aguycalled) [inline screens/wallet/SettingsScreen.tsx:179]: is this needed?

**Pull request merged**